### PR TITLE
fix: 198 broken border

### PR DESCRIPTION
**Pull request by ArthurEncr**
# Description

fixing the bug that when the page height is decreased and the scrollbar appears, the borders of the panel section name keep showing.

Fixes #[198](https://github.com/bcgov/nr-spar/issues/198)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Manualy tested.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-291-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-291-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-291-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/components/PanelSectionName/styles.scss [hunk 1]
```diff
@@ -27,3 +27,10 @@
   color: var(--#{vars.$bcgov-prefix}-text-secondary);
   user-select: none;
 }
+
+@media only screen and (max-height: 877px) {
+  .panel-section,
+  .panel-section-light {
+  box-shadow: none
+}
```
Comment (mgaseta) [inline frontend/src/components/PanelSectionName/styles.scss:35]: Adjust indentation here.

File: frontend/src/components/PanelSectionName/styles.scss [hunk 2]
```diff
@@ -27,3 +27,10 @@
   color: var(--#{vars.$bcgov-prefix}-text-secondary);
   user-select: none;
 }
+
+@media only screen and (max-height: 877px) {
```
Comment (mgaseta) [inline frontend/src/components/PanelSectionName/styles.scss:31]: Add this `max-height` value to a var on `variables.scss` file, something like left-menu-height and use it in this case.

**Review by mgaseta — CHANGES_REQUESTED** [review]
Nice job! :rocket: Added only 2 small markups.

File: frontend/src/components/PanelSectionName/styles.scss (hunk 2 above)
Comment (craigyu) [reply frontend/src/components/PanelSectionName/styles.scss:31]: @mgaseta are you referring to `frontend/src/theme/variables.scss`? we will be using this [nr-theme package](https://github.com/bcgov/nr-fsa-theme/tree/main) so we should not make edit to this file

File: frontend/src/components/PanelSectionName/styles.scss [hunk 3]
```diff
@@ -27,3 +27,10 @@
...
+  .panel-section,
+  .panel-section-light {
+  box-shadow: none
+}
+}
```
Comment (craigyu) [inline frontend/src/components/PanelSectionName/styles.scss:36]: missing empty line

**Comment by craigyu** [comment]
@mgaseta @ArthurEncr 
The root cause of the missing border issue is caused by the `component-override.scss` file, because it's using `box-shadow` for some reason, changing it to `border-right` seem to have fixed the problem. 

Sorry to hijack your PR Arthur, I made a new commit [here](https://github.com/bcgov/nr-spar/pull/291/commits/a72c1c80234673f403028d8ca69fffab3cb063f0).

### Additional changes:
- replaced hard-coded css prefix
- replaced text styling by taking advantage of `@use '@carbon/type';`

**Review by craigyu — APPROVED** [review]

File: frontend/src/components/PanelSectionName/styles.scss (hunk 2 above)
Comment (mgaseta) [reply frontend/src/components/PanelSectionName/styles.scss]: Forgot about that :sweat_smile: , I'll close this markup.

File: frontend/src/theme/components-overrides.scss [hunk 4]
```diff
@@ -125,7 +125,7 @@
 }
 
 nav.#{variables.$bcgov-prefix}--side-nav {
-  box-shadow: inset -1px 0 0 colors.$gray-20;
+  border-right: 1px solid colors.$gray-20;
   background-color: var(--bcgov-background-left-panel);
```
Comment (mgaseta) [inline frontend/src/theme/components-overrides.scss:129]: Use the bcgov prefix variable here.

**Review by mgaseta — CHANGES_REQUESTED** [review]
Nice changes! Added only one small markup.

File: frontend/src/theme/components-overrides.scss (hunk 4 above)
Comment (craigyu) [reply frontend/src/theme/components-overrides.scss:129]: we actually found many hardcoded prefix like this one yesterday during an nr-theme meeting. I will be creating an issue to tackle all of them so we can start using the nr-theme package

Comment (craigyu) [reply frontend/src/theme/components-overrides.scss:129]: actually I'll just fix it here

Comment (craigyu) [reply frontend/src/theme/components-overrides.scss]: [fixed](https://github.com/bcgov/nr-spar/pull/291/commits/dcd20052f5634728bd69f9310b08701d675bbab4)

**Pull request merged**